cli/src/main/kotlin/commands/UploadResultToSw360Command.kt

```diff
@@ -89,9 +107,27 @@ class UploadResultToSw360Command : CliktCommand(
         getProjectWithPackages(ortResult).forEach { (project, pkgList) ->
             val linkedReleases = pkgList.mapNotNull { pkg ->
                 val name = createReleaseName(pkg.id)
-                sw360ReleaseClient.getSparseReleaseByNameAndVersion(name, pkg.id.version)
+                var release = sw360ReleaseClient.getSparseReleaseByNameAndVersion(name, pkg.id.version)
                     .flatMap { sw360ReleaseClient.enrichSparseRelease(it) }
                     .orElseGet { createSw360Release(pkg, sw360ReleaseClient) }
+
+                if (attachSources) {
+                    val tempDirectory = createTempDirectory(pkg.id.toCoordinates()).toFile()
+                    val saveSourcesTemporarily = saveSourcesTemporarily(pkg, tempDirectory)
+
+                    saveSourcesTemporarily?.let {
+                        release = uploadAttachmentToRelease(
```

Comment:
Just curious: Is the `release` returned here really any different from the passed-in `release`? I would have assumed that attaching sources does not change the release per se. Because if it doesn't, we could simplify the code here a bit.

Reply:
It actually is, as it has been replaced by the instance returned from the server, which now contains the attachment file. That said, given the way the release is later used (just to establish the link to the project), this difference should not really matter, as only the release ID is relevant. So I think it should be safe to stick to the original release instance.

Reply:
Yes, it's different, but I just noticed that it doesn't matter because only the `releaseId` is relevant to for the further run.